caas/kubernetes/provider/specs/types.go

```diff
@@ -205,3 +206,16 @@ func getParser(specVersion specs.Version) (parserType, error) {
 		return nil, errors.NewNotSupported(nil, fmt.Sprintf("latest supported version %d, but got podspec version %d", specs.CurrentVersion, specVersion))
 	}
 }
+
+// IntOrStringToK8s converts IntOrString to k8s version.
+func IntOrStringToK8s(in specs.IntOrString) (*intstr.IntOrString, error) {
+	data, err := in.MarshalJSON()
```

Comment:
I wonder why we can't just copy the fields into a result struct?
Less expensive than marshalling?

Reply:
right